Lib/test/datetimetester.py

```diff
@@ -1363,6 +1364,14 @@ def test_isocalendar(self):
             d = self.theclass(2010, 1, 4+i)
             self.assertEqual(d.isocalendar(), (2010, 1, i+1))
 
+    def test_isocalendar_named(self):
```

Comment:
How about refactoring this test and `test_isocalendar` together into a single parametrized test:

```python
def test_isocalendar(self):
    # Check examples from
    # https://www.phys.uu.nl/~vgent/calendar/isocalendar.htm
    week_mondays = [
        ((2003, 12, 22), (2003, 52, 1)),
        ((2003, 12, 29), (2004, 1, 1)),
        ((2004, 1, 5), (2004, 2, 1)),
        ((2009, 12, 21), (2009, 52, 1)),
        ((2009, 12, 28), (2009, 53, 1)),
        ((2010, 1, 4), (2010, 1, 1)),
    ]

    test_cases = []
    for cal_date, iso_date in week_mondays:
        base_date = self.theclass(*cal_date)

        # Adds one test case for every day of the specified weeks
        for i in range(7):
            new_date = base_date + timedelta(i)
            new_iso = iso_date[0:2] + (iso_date[2] + 1,)

            test_cases.append((new_date, new_iso))

    for d, exp_iso in test_cases:
        with self.subTest(d=d, comparison="tuple"):
            self.assertEqual(d.isocalendar(), exp_iso)

        # Check that the tuple contents are accessible by field name
        with self.subTest(d=d, comparison="fields"):
            iso_tuple = d.isocalendar()
            self.assertEqual((iso_tuple.year, iso_tuple.week, iso_tuple.day),
                             exp_iso)
```

I recognize that it adds a bit of complexity to the code, but it will give more granular output than we have now while also allowing us to easily add new examples if needed.

Reply:
Great idea